### ISSUE TITLE
Progress bars are not showing in the end

### ISSUE THREAD

**Issue by Qazalbash** [Textualize/rich#3685]
Hi, I train neural networks and want to see their loss in each epoch. I have set a progress bar for this. When the progress bar is over the height of the terminal it stops showing progress bars for new epochs. I have attached the reproducible example. Let me know if it is fixable or a bug.

```python
import random
import time

from rich.progress import (
    BarColumn,
    MofNCompleteColumn,
    Progress,
    SpinnerColumn,
    TextColumn,
    TimeRemainingColumn,
)


total = 100
epochs = 70

with Progress(
    SpinnerColumn(),
    TextColumn("Epoch {task.fields[epoch]}"),
    BarColumn(),
    MofNCompleteColumn(),
    TimeRemainingColumn(elapsed_when_finished=True),
    TextColumn("[progress.description]{task.description}"),
    refresh_per_second=5,
) as progress:
    for epoch in range(epochs):
        task_id = progress.add_task(
            "",
            total=total,
            epoch=epoch + 1,
        )
        for i in range(0, 100):
            loss = random.uniform(0, 1)
            progress.update(
                task_id,
                advance=1,
                description=f"loss: {loss:.4f}",
            )

        val_loss = random.uniform(0, 1)
        progress.update(
            task_id,
            completed=total,
            refresh=True,
            description=f"loss: {loss:.4f} - val loss: {val_loss:.4f}",
        )
        time.sleep(0.1)
```

![Image](https://github.com/user-attachments/assets/9fdb5e7e-92de-42be-8842-43a097378e9a)

**Comment by willmcgugan** [comment]
If you have 70 tasks and less that 70 lines, you aren't going to be able to display all of them at once. Some are going to be hidden.

You could remove tasks that are completed, but at this point I think you should consider something more interactive, like Textual.

**Comment by Qazalbash** [comment]
> If you have 70 tasks and less that 70 lines, you aren't going to be able to display all of them at once. Some are going to be hidden.

My intuition was I would be able to scroll up or down. Is that not possible in the current state?

> You could remove tasks that are completed, but at this point I think you should consider something more interactive, like Textual.

Removing them means I will not be able to see the evolution of loss. I will explore Textual, thank you!

**Comment by willmcgugan** [comment]
Rich is generally output-only. Textual is the way to go if you want to scroll a list, or arrange the scrollbars more efficiently.

**Comment by Qazalbash** [comment]
Thanks! I will explore textual then.
